modules/indexExchangeBidAdapter.js

```diff
@@ -425,132 +488,36 @@ var IndexExchangeAdapter = function IndexExchangeAdapter() {
-          if (!(bid.sizes[j][0] == bid.params.size[0] && bid.sizes[j][1] == bid.params.size[1])) {
-            passOnBid(bid.placementCode);
-            continue;
-          }
-          usingSizeSpecificSiteID = true;
-        }
-
-        if (bid.params.timeout && typeof cygnus_index_args.timeout === 'undefined') {
-          cygnus_index_args.timeout = bid.params.timeout;
-        }
-
-        var siteID = Number(bid.params.siteID);
-        if (typeof siteID !== 'number' || siteID % 1 != 0 || siteID <= 0) {
-          utils.logMessage(ADAPTER_NAME + ' slot excluded from request due to invalid siteID');
-          passOnBid(bid.placementCode);
-          continue;
-        }
-        if (siteID && typeof cygnus_index_args.siteID === 'undefined') {
-          cygnus_index_args.siteID = siteID;
-        }
-
-        if (utils.hasValidBidRequest(bid.params, requiredParams, ADAPTER_NAME)) {
-          firstAdUnitCode = bid.placementCode;
-          var slotID = bid.params[requiredParams[0]];
-          if (typeof slotID !== 'string' && typeof slotID !== 'number') {
-            utils.logError(ADAPTER_NAME + ' bid contains invalid slot ID from ' + bid.placementCode + '. Discarding slot');
-            passOnBid(bid.placementCode);
-            continue
-          }
-
-          sizeID++;
-          var size = {
-            width: bid.sizes[j][0],
-            height: bid.sizes[j][1]
-          };
-
-          var slotName = usingSizeSpecificSiteID ? String(slotID) : slotID + '_' + sizeID;
-          slotIdMap[slotName] = bid;
-
-          // Doesn't need the if(primary_request) conditional since we are using the mergeSlotInto function which is safe
-          cygnus_index_args.slots = mergeSlotInto({
-            id: slotName,
-            width: size.width,
-            height: size.height,
-            siteID: siteID || cygnus_index_args.siteID
-          }, cygnus_index_args.slots);
-
-          if (bid.params.tier2SiteID) {
-            var tier2SiteID = Number(bid.params.tier2SiteID);
-            if (typeof tier2SiteID !== 'undefined' && !tier2SiteID) {
-              continue;
-            }
-
-            cygnus_index_args.slots = mergeSlotInto({
-              id: 'T1_' + slotName,
-              width: size.width,
-              height: size.height,
-              siteID: tier2SiteID
-            }, cygnus_index_args.slots);
-          }
-
-          if (bid.params.tier3SiteID) {
-            var tier3SiteID = Number(bid.params.tier3SiteID);
-            if (typeof tier3SiteID !== 'undefined' && !tier3SiteID) {
-              continue;
-            }
-
-            cygnus_index_args.slots = mergeSlotInto({
-              id: 'T2_' + slotName,
-              width: size.width,
-              height: size.height,
-              siteID: tier3SiteID
-            }, cygnus_index_args.slots);
-          }
+    // Grab the slot level data for cygnus_index_args
+    for (var i = 0; i < bidArr.length; i++) {
+      let bid = bidArr[i];
```

Comment:
if you're just iterating over an array for references to each item, you should use `bidArr.forEach(bid =>{})`